src/dsp_tools/commands/excel2json/json_header.py

```diff
@@ -0,0 +1,249 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any
+from typing import cast
+
+import pandas as pd
+import regex
+
+from dsp_tools.commands.excel2json.models.input_error import AtLeastOneValueRequiredProblem
+from dsp_tools.commands.excel2json.models.input_error import EmptySheetsProblem
+from dsp_tools.commands.excel2json.models.input_error import ExcelFileProblem
+from dsp_tools.commands.excel2json.models.input_error import ExcelSheetProblem
+from dsp_tools.commands.excel2json.models.input_error import InvalidExcelContentProblem
+from dsp_tools.commands.excel2json.models.input_error import MissingValuesProblem
+from dsp_tools.commands.excel2json.models.input_error import MoreThanOneRowProblem
+from dsp_tools.commands.excel2json.models.input_error import PositionInExcel
+from dsp_tools.commands.excel2json.models.input_error import Problem
+from dsp_tools.commands.excel2json.models.input_error import RequiredColumnMissingProblem
+from dsp_tools.commands.excel2json.models.json_header import Descriptions
+from dsp_tools.commands.excel2json.models.json_header import EmptyJsonHeader
+from dsp_tools.commands.excel2json.models.json_header import FilledJsonHeader
+from dsp_tools.commands.excel2json.models.json_header import JsonHeader
+from dsp_tools.commands.excel2json.models.json_header import Keywords
+from dsp_tools.commands.excel2json.models.json_header import Prefixes
+from dsp_tools.commands.excel2json.models.json_header import Project
+from dsp_tools.commands.excel2json.models.json_header import User
+from dsp_tools.commands.excel2json.models.json_header import UserRole
+from dsp_tools.commands.excel2json.models.json_header import Users
+from dsp_tools.commands.excel2json.utils import check_contains_required_columns
+from dsp_tools.commands.excel2json.utils import check_required_values_get_position_in_excel
+from dsp_tools.commands.excel2json.utils import read_and_clean_all_sheets
+from dsp_tools.models.exceptions import InputError
+
+
+def get_json_header(excel_filepath: Path) -> JsonHeader:
+    """
+    Returns the header of the JSON file.
+    If an Excel file with the information is provided it is filled out.
+    Otherwise it will return a header with the fields left blank.
+
+    Args:
+        excel_filepath: path to the excel file
+
+    Returns:
+        JsonHeader object
+
+    Raises:
+        InputError: If the file does not conform to the specifics
+    """
+    if not excel_filepath.is_file():
+        print("No json_header.xlsx file found in the directory, empty header was added.")
+        return EmptyJsonHeader()
+    sheets_df_dict = read_and_clean_all_sheets(excel_filepath)
+    sheets_df_dict = {x.lower(): df for x, df in sheets_df_dict.items()}
```

Comment:
this seems rather low-level. Could this step simply be integrated in the function above?

Reply:
No I would not do that by default. We use this function for all excel2json, including the classes where the names are relevant. So we would have to change some code there too...

Reply:
fair enough, I was assuming this function was just for that purpose